assemblies/static/src/main/resources/static/custom/js/sparkl.js

```diff
@@ -1,423 +1,454 @@
-                type: 'GET',
-                dataType: opts.dataType,
-                success: successHandler,
-                error: errorHandler,
-                data: {}
-            };
-        }
+        $submitInput.click();
+      }
+    });
 
-        _.each(opts.params, function(value, key) {
-            ajaxOpts.data['param' + key] = value;
-        });
-        _.each(opts.systemParams, function(value, key) {
-            ajaxOpts.data[key] = value;
-        });
+    return function () {
+      $fileInput.click();
+    };
+  };
 
-        $.ajax(ajaxOpts);
+  myself.testFile = function (url, successCallback, errorCallback) {
+    successCallback = successCallback || function () {
+      return true;
     };
 
-    myself.getEndpointCaller = function(pluginId, endpoint, opts) {
-        var myself = this;
-        return function(callback, errorCallback, params) {
-            var _opts = $.extend({}, opts);
-            _opts.params = params || _opts.params;
-            _opts.success = callback || _opts.success;
-            _opts.error = errorCallback || _opts.error;
-            myself.runEndpoint(pluginId, endpoint, _opts);
-        };
+    errorCallback = errorCallback || function () {
+      return true;
     };
 
-    myself.publishToServer = function(callback) {
-        $.ajax({
-            url: Dashboards.getWebAppPath() + '/plugin/sparkl/api/reloadPlugins',
-            type: 'POST',
-            data: {
-                'publish': 'now',
-                'class': 'org.pentaho.platform.plugin.services.pluginmgr.PluginAdapter'
-            },
-            success: callback
-        });
+    $.ajax({
+      url: url,
+      type: 'HEAD',
+      error: errorCallback,
+      success: successCallback
+    });
+  };
+
+
+  myself.runEndpoint = function (pluginId, endpoint, opts0) {
+    if (!pluginId && !endpoint) {
+      Dashboards.log('PluginId or endpointName not defined.');
+
+      return false;
+    }
+
+    var _opts = {
+      success: function () {
+        Dashboards.log(pluginId + ': ' + endpoint + ' ran successfully.');
+      },
+
+      error: function () {
+        Dashboards.log(pluginId + ': error running ' + endpoint + '.');
+      },
+
+      params: {},
+      systemParams: {},
+
+      type: 'POST',
+      dataType: 'json'
     };
 
-    myself.addCallWrapper = function(caller, callback) {
-        return function(json) {
-            caller(callback);
-        };
+    var opts = $.extend({}, _opts, opts0);
+    var url = Dashboards.getWebAppPath() + '/plugin/' + pluginId + '/api/' + endpoint;
+
+    function successHandler(json) {
+      if (json && json.result === false) {
+        opts.error.apply(this, arguments);
+      } else {
+        opts.success.apply(this, arguments);
+      }
+    }
+
+    function errorHandler() {
+      opts.error.apply(this, arguments);
+    }
+
+    var ajaxOpts = {
+      url: url,
+      async: true,
+      type: endpoint !== 'renderer/refresh' ? opts.type : 'GET',
```

Comment:
Should you preserve the "do this better comment"?

Reply:
I removed it because this is a bit better than the previous version, and the chance of active development on this plugin is close very to 0%. So it feels like this comment is no longer necessary